applications/API/serializers.py

```
@@ -0,0 +1,574 @@
+        Raises:
+            serializers.ValidationError: If the category cannot be resolved by key or name.
+        """
+        if not value:
+            return {"category_key": "", "category_name": "Uncategorized"}
+
+        if isinstance(value, InsultCategory):
+            validated = {
+                "category_key": value.category_key,
+                "category_name": value.name,
+            }
+            logger.debug(f"Serializer Resolved {value} to {validated}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)